<<<<<<< HEAD
---
description: 'Require `Array#sort` calls to always provide a `compareFunction`.'
---

> 🛑 This file is source code, not the primary documentation location! 🛑
>
> See **https://typescript-eslint.io/rules/require-array-sort-compare** for documentation.

When called without a compare function, `Array#sort()` converts all non-undefined array elements into strings and then compares said strings based off their UTF-16 code units [[ECMA specification](https://www.ecma-international.org/ecma-262/9.0/#sec-sortcompare)].

The result is that elements are sorted alphabetically, regardless of their type.
For example, when sorting numbers, this results in a "10 before 2" order:
=======
# Enforce giving `compare` argument to `Array#sort` (require-array-sort-compare)

This rule prevents to invoke `Array#sort()` method without `compare` argument.

`Array#sort()` method sorts that element by the alphabet order.
>>>>>>> 491790fa3284842aea531bfa9d2fd819794b4f37

```ts
[1, 2, 3, 10, 20, 30].sort(); //→ [1, 10, 2, 20, 3, 30]
```

<<<<<<< HEAD
This rule reports on any call to the `Array#sort()` method that doesn't provide a `compare` argument.

## Examples

This rule aims to ensure all calls of the native `Array#sort` method provide a `compareFunction`, while ignoring calls to user-defined `sort` methods.

<!--tabs-->

### ❌ Incorrect
=======
The language specification also noted this trap.

> NOTE 2: Method calls performed by the ToString abstract operations in steps 5 and 7 have the potential to cause SortCompare to not behave as a consistent comparison function.<br> > https://www.ecma-international.org/ecma-262/9.0/#sec-sortcompare

## Rule Details

This rule is aimed at preventing the calls of `Array#sort` method.
This rule ignores the `sort` methods of user-defined types.

Examples of **incorrect** code for this rule:
>>>>>>> 491790fa3284842aea531bfa9d2fd819794b4f37

```ts
const array: any[];
const stringArray: string[];

array.sort();

<<<<<<< HEAD
// String arrays should be sorted using `String#localeCompare`.
stringArray.sort();
```

### ✅ Correct
=======
// Even if a string array, warns it in favor of `String#localeCompare` method.
stringArray.sort();
```

Examples of **correct** code for this rule:
>>>>>>> 491790fa3284842aea531bfa9d2fd819794b4f37

```ts
const array: any[];
const userDefinedType: { sort(): void };

array.sort((a, b) => a - b);
array.sort((a, b) => a.localeCompare(b));

userDefinedType.sort();
```

<<<<<<< HEAD
## Options

### `ignoreStringArrays`

Examples of code for this rule with `{ ignoreStringArrays: true }`:

<!--tabs-->

#### ❌ Incorrect

```ts
const one = 1;
const two = 2;
const three = 3;
[one, two, three].sort();
```

#### ✅ Correct

```ts
const one = '1';
const two = '2';
const three = '3';
[one, two, three].sort();
```

## When Not To Use It

If you understand the language specification enough, and/or only ever sort arrays in a string-like manner, you can turn this rule off safely.
=======
### Options

There is no option.

## When Not To Use It

If you understand the language specification enough, you can turn this rule off safely.
>>>>>>> 491790fa3284842aea531bfa9d2fd819794b4f37
